### ISSUE TITLE
Running tests produces spurious output

### ISSUE THREAD

**Issue by teknico** [decred/dcrutil#60]
Running tests produces spurious output from `go vet`:
```
$ ./run_tests.sh local
[...]
bloom/example_test.go:20::error: ExampleNewFilter refers to unknown identifier: \
  NewFilter (vet)
hdkeychain/example_test.go:17::error: ExampleNewMaster refers to unknown identifier: \
  NewMaster (vet)
++ test -z 'bloom/example_test.go:20::error: ExampleNewFilter refers to unknown \
  identifier: NewFilter (vet)
hdkeychain/example_test.go:17::error: ExampleNewMaster refers to unknown identifier: \
  NewMaster (vet)'
```
This is due to a [bug](https://github.com/alecthomas/gometalinter/issues/315) in the way `gometalinter` invokes `go tool vet`.

Another project [worked around it](https://github.com/01org/ciao/pull/1321) by using a tagged version of `gometalinter`.

**Comment by jolan** [comment]
https://github.com/alecthomas/gometalinter/issues/218 looks more relevant.

Renaming the example function as done in #62 works locally for me with go 1.9.

**Comment by teknico** [comment]
> alecthomas/gometalinter#218 looks more relevant.

It indeed is.

> Renaming the example function as done in #62 works locally for me with go 1.9.

It does work, how weird. That's unpredictable behavior on the part of `go vet`.

You would also have to make the parallel change to `hdkeychain`'s `example_test.go` file, though.

**Comment by davecgh** [comment]
Fixed by decred/dcrd#890 since the code has been moved into dcrd proper now.
